android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -2206,11 +2254,25 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
         editor.putBoolean(KMManager.KMKey_ShouldShowHelpBubble, false);
         editor.commit();
 
+        // Update globeKeyState
+        int start = url.indexOf("keydown=") + 8;
+        String value = url.substring(start);
+        boolean globeKeyDown = !value.isEmpty() && Boolean.valueOf(value);
```

Comment:
I'm seeing this url parsing pattern over and over again. I'd like to see us using a URL parser class instead and then having a helper function if necessary, e.g.

```java
  boolean globeKeyDown = getUrlBooleanValue("keydown");
```

This removes the unnecessary local variable definitions of `start` and `value`. 

In fact, even better would be to extract the entirety of the if test for url.indexOf("globeKeyAction") to a function:

```java
  } else if (url.indexOf("globeKeyAction") >= 0) {
    handleGlobeKeyAction(getUrlBooleanValue(url, "keydown"));
  }
```